src/pages/index.js

```diff
@@ -0,0 +1,327 @@
+    if (cardToDelete) {
+      const cardId = cardToDelete.dataset.cardId;
+
+      api.deleteCard(cardId)
+        .then(() => {
+          cardToDelete.remove();
+          closeModal(deleteModal);
+        })
+        .catch(err => {
+          console.error('Error deleting card:', err);
+        });
+    }
+  }
+
+
+  // Cards
+  const cardTemplate = document.querySelector('#card-template').content;
+  const cardsContainer = document.querySelector('.cards__list');
+
+  api.getAppInfo()
+    .then(([userInfo, cards]) => {
+      document.querySelector('.profile__name').textContent = userInfo.name;
+      document.querySelector('.profile__description').textContent = userInfo.about;
+      document.getElementById('profile-avatar').src = userInfo.avatar;
+
+      renderInitialCards(cards);
+    })
+    .catch(err => {
+      console.error('Error receiving user information:', err);
+    });
+
+  function renderInitialCards(cards) {
+    cards.forEach((card) => {
+      const cardElement = getCardElement(card);
+      cardsContainer.appendChild(cardElement);
+    });
+  }
+
+  function getCardElement(data) {
+    const cardElement = cardTemplate.cloneNode(true).querySelector('.card');
+    const cardImage = cardElement.querySelector('.card__image');
+    const cardTitle = cardElement.querySelector('.card__title');
+    const cardLikeBtn = cardElement.querySelector('.card__like-btn');
+    const cardDeleteBtn = cardElement.querySelector('.card__delete-btn');
+
+    cardElement.dataset.cardId = data._id;
+
+
+    if (data.isLiked) {
+      cardLikeBtn.classList.add('card__like-btn_liked');
+    } else {
+      cardLikeBtn.classList.remove('card__like-btn_liked');
+    }
+
+    cardLikeBtn.addEventListener('click', () => {
+      if (cardLikeBtn.classList.contains('card__like-btn_liked')) {
+        api.unlikeCard(data._id)
+          .then(updatedCard => {
+            cardLikeBtn.classList.remove('card__like-btn_liked');
+          })
+          .catch(err => console.error('Error unliking card:', err));
+      } else {
+        api.likeCard(data._id)
+          .then(updatedCard => {
+            cardLikeBtn.classList.add('card__like-btn_liked');
+          })
+          .catch(err => console.error('Error liking card:', err));
+      }
+    });
+
+    cardImage.addEventListener('click', () => {
+      openModal(previewModal);
+      previewModalImage.src = data.link;
+      previewModalImage.alt = data.name;
+      previewModalCaption.textContent = data.name;
+    });
+
+    cardDeleteBtn.addEventListener('click', () => {
+      openDeleteModal(cardElement);
+    });
+
+    cardImage.src = data.link;
+    cardImage.alt = data.name;
+    cardTitle.textContent = data.name;
+
+    return cardElement;
+  }
+
+
+
+  function handleEditProfileSubmit(evt) {
+    evt.preventDefault();
+    const name = editModalNameInput.value;
+    const about = editModalDescriptionInput.value;
+
+    const saveButton = editProfileForm.querySelector('.modal__submit-btn');
```

Comment:
**Could be improved**


No need in seraching the button because `evt.submitter` is the button on submitting

```js
    const saveButton = evt.submitter
```